logging/scripts/shiplog.sh

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env bash
+#
+# logging/scripts/shiplog.sh
+#
+# Ship today's log entries from a specified log to logz.io
+# Usage: /bin/bash /app/logging/scripts/shiplog.sh [LOG_NAME] [LOG_PATH] [LOG_DATE_PATTERN] [LOG_TYPE]
+#    eg: /bin/bash /app/logging/scripts/shiplog.sh "access" "/var/log/access.log" "$(date +%d/%b/%Y:)" "nginx_access"'
+
+echo "Shipping log ..."
+
+# LOGZ_TOKEN environment variable is required for this script to run.
+if [ -z "$LOGZ_TOKEN" ]; then
+    echo "LOGZ_TOKEN is not set - exiting"
+    exit 1
+fi
+
+# Mount for logs must exist or exit script.
+cd /app/log || exit
+
+LOG_NAME=$1
+LOG_PATH=$2
+LOG_DATE_PATTERN=$3
+LOG_TYPE=$4
+
+# Script will be creating logs for today and removing yesterday's logs
+TODAY_DATE=$(date +%Y-%m-%d)
+YESTERDAY_DATE=$(date --date="yesterday" +%Y-%m-%d)
+
+if [ -z "$LOG_NAME" ] || [ -z "$LOG_PATH" ] || [ -z "$LOG_DATE_PATTERN" ] || [ -z "$LOG_TYPE" ]; then
```

Comment:
This is grand and just  a side not that you can use $# to check the number of params passed

if [ $# -ne 4 ]
  then
    echo "This command requires 4 parameters, $# passed"
fi

Reply:
Ahhhh ok - nice one 👍🏻